### PR TITLE
WIP: Update ont-genexis-ft-g2110c.md

### PR DESCRIPTION
**Pull request by stich86**
Adding useful information

### CODE REVIEW

File: _ont/ont-genexis-ft-g2110c.md [hunk 1]
```diff
@@ -65,6 +66,15 @@ This ONT supports dual boot.
 
 # Useful Commands
 
+Change GPON Serial Number:
+
```
Comment (simonebortolin) [inline _ont/ont-genexis-ft-g2110c.md:70]: add ```sh

File: _ont/ont-genexis-ft-g2110c.md [hunk 2]
```diff
@@ -65,6 +66,15 @@ This ONT supports dual boot.
...
+set GponSn GNXS05542100
+save env
+
```
Comment (simonebortolin) [inline _ont/ont-genexis-ft-g2110c.md:73]: add ```

File: _ont/ont-genexis-ft-g2110c.md [hunk 3]
```diff
@@ -65,6 +66,15 @@ This ONT supports dual boot.
...
+Change ONT IP Address:
+
+flash set LAN_IP_ADDR 192.168.1.1
```
Comment (simonebortolin) [inline _ont/ont-genexis-ft-g2110c.md:76]: add ```sh

File: _ont/ont-genexis-ft-g2110c.md [hunk 4]
```diff
@@ -65,6 +66,15 @@ This ONT supports dual boot.
...
+
```
Comment (simonebortolin) [inline _ont/ont-genexis-ft-g2110c.md:77]: add ```

**Review by simonebortolin — CHANGES_REQUESTED** [review]

**Review by simonebortolin — APPROVED** [review]

File: _ont/ont-genexis-ft-g2110c.md [hunk 5]
```diff
@@ -65,7 +69,59 @@ This ONT supports dual boot.
...
+```sh
+# nv setenv ploampwd 1234567890
+ploampwd=1234567890
+```
 
 # Known Bugs
+
+- ONT Serial Number can only be changed via U-Boot console, if you try to change it from the OS you will receive an error
+- Most of value on XML configuration are not readed by `omci_app` deamon, so changing parameters like `PON_VENDOR_ID`, `OLT_MODE` or `SW_VER*` are not supported. A the moment this unit cannot be fully spoofed like other Realtek's ONT with stock Luna SDK
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:124]:
```suggestion
- Most of the values inside the XML configuration file are not read by the `omci_app` deamon, so changing parameters like `PON_VENDOR_ID`, `OLT_MODE` or `SW_VER*` is not supported. At the moment this unit cannot be fully spoofed like other Realtek's ONT with stock Luna SDK
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 6]
```diff
@@ -65,7 +69,59 @@ This ONT supports dual boot.
 
 # Useful Commands
 
+## Getting/Setting the ONT's S/N
+
+{% include alert.html content="The GPON S/N can only be changed via U-Boot console" alert="Note"  icon="svg-info" color="blue" %}
+
+Attach TTL adapter to UART console (see Internal photo - CM3 header, for pinout), open a terminal emulator (like Putty or Teraterm), power on the ONT and `CTRL+C` until you see the following prompt:
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:76]:
```suggestion
Attach a TTL adapter to the UART console port (see Internal photo - CM3 header, for pinout), open a terminal emulator (like Putty or Teraterm), power on the ONT and press `CTRL+C` until you see the following prompt:
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 7]
```diff
@@ -65,7 +69,59 @@ This ONT supports dual boot.
...
+
+```sh
+9601D#
+```
+
+On that console you can type the following commands to change the serial number of ONT:
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:82]:
```suggestion
On that console you can type the following commands to change the GPON serial number:
```

**Review by nanomad — CHANGES_REQUESTED** [review]

File: _ont/ont-genexis-ft-g2110c.md [hunk 8]
```diff
@@ -65,7 +69,59 @@ This ONT supports dual boot.
...
+```
+
+You can check if the serial number was correclty changed using the following command:
+
+```sh
+# diag gpon get serialnumber
+GNXS05542100
+```
+
+## Change IP address
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:97]:
```suggestion
## Changing the IP address
```

**Review by nanomad — CHANGES_REQUESTED** [review]

**Review by ErnyTech — APPROVED** [review]

File: _ont/ont-genexis-ft-g2110c.md [hunk 9]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+Committed: 1
+Valid: 1
+*Version: V6.0.10N14*
+ProductCode:
+ImageHash: 0x00000000000000000000000000000000
+=================================
+```
 
 # Known Bugs
+- Modified image (based on C-5.7.1-DEV3) has Web Gui enabled, but without attached fiber the daemon will crash
```
Comment (simonebortolin) [inline _ont/ont-genexis-ft-g2110c.md:179]: in what sense does it crash?

File: _ont/ont-genexis-ft-g2110c.md [hunk 10]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+## Getting/Setting the ONT's PLOAM password
+
+{% include alert.html content="The PLOAM password is stored in ASCII format" alert="Info" icon="svg-info" color="blue" %}
+
+```sh
+# /etc/scripts/flash get GPON_PLOAM_PASSWD
+GPON_PLOAM_PASSWD=AAAAAAAAAA
+# /etc/scripts/flash set GPON_PLOAM_PASSWD AAAAAAAAAA
+```
+Looks like Genexis has also included an U-Boot variable to store the PLOAM. On currently known firmwares (5.6.1 and 5.7.0) it is not read by the `omci_app` daemon.
```
[the remark this answers is not in the document]
Comment (stich86) [reply _ont/ont-genexis-ft-g2110c.md:127]: ATM is the only one used. The Uboot env is there but omci_app doesn't read it, may be in the next update... who knows :)

File: _ont/ont-genexis-ft-g2110c.md (hunk 9 above)
Comment (stich86) [reply _ont/ont-genexis-ft-g2110c.md:179]: If you call the web page without fiber attached, you will get the first page and the a server error and the boa daemon dies, so only way to restore http server is to start manually or reboot the whole ONT

Comment (simonebortolin) [reply _ont/ont-genexis-ft-g2110c.md:179]: OK maybe it should be written better

File: _ont/ont-genexis-ft-g2110c.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (simonebortolin) [reply _ont/ont-genexis-ft-g2110c.md:127]: ok

File: _ont/ont-genexis-ft-g2110c.md [hunk 11]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+
+Use the following command to unlock write-protected variables:
+
+```sh
+# nv setenv ProtectEnv 0
+```
+
+## Getting/Setting the ONT's S/N
+
+{% include alert.html content="If your serial has letters after the VID (ex. ZTEGAAA01234), put them in lower case (ex. ZTEGaaa01234)" alert="Info" icon="svg-info" color="blue" %}
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:83]:
```suggestion
{% include alert.html content="If your serial number has letters after the Vendor ID (e.g. ZTEGAAA01234), put them in lower case (e.g. ZTEGaaa01234)" alert="Info" icon="svg-info" color="blue" %}
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 12]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+
+```sh
+# nv setenv GponSn ZTEGaaa01234
+```
+Reboot ONT to apply the change
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:88]:
```suggestion
Reboot the ONT to apply any changes
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 13]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+
+{% include alert.html content="The PLOAM password is stored in ASCII format" alert="Info" icon="svg-info" color="blue" %}
+
+```sh
+# /etc/scripts/flash get GPON_PLOAM_PASSWD
+GPON_PLOAM_PASSWD=AAAAAAAAAA
+# /etc/scripts/flash set GPON_PLOAM_PASSWD AAAAAAAAAA
+```
+Looks like Genexis has also included an U-Boot variable to store the PLOAM. On currently known firmwares (5.6.1 and 5.7.0) it is not read by the `omci_app` daemon.
+If you want to be future proof, put your PLOAM also on U-Boot env using the following command on the OS shell:
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:128]:
```suggestion
If you want to be future proof, put your PLOAM also on U-Boot env using the following command in the OS shell:
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 14]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+
+```sh
+# nv setenv ploampwd 1234567890
+ploampwd=1234567890
+```
+
+## Spoofing firmware version
+
+{% include alert.html content="To change this information you need to upload a modified firmware. Do it at your own risk!" alert="Warning" icon="svg-warning" color="red" %}
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:137]:
```suggestion
{% include alert.html content="To change this data you need to upload a modified firmware. Do it at your own risk!" alert="Warning" icon="svg-warning" color="red" %}
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 15]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+
+- Download the modified firmware from [here](https://mega.nz/file/yspFRTJT#JjeDS7NwLZxCzFklQGMJVIuPawo8utQRGbMGEXQv84M) md5:  `91946ad8aada2e469b7fff1d38ee7474`
+- Logon on OS shell of the ONT and run this command: `iptables -F`
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:140]:
```suggestion
- Logon on the OS shell of the ONT and run this command: `iptables -F`
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 16]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+- Open a web browser and go to `http://192.168.1.1/upgrade.asp` or `http://192.168.100.1/upgrade.asp`, enter same credentials used for telnet session, select the modified firmware and click on `Upgrade`
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:141]:
```suggestion
- Open a web browser and go to `http://192.168.1.1/upgrade.asp` or `http://192.168.100.1/upgrade.asp`, enter the same credentials used for telnet session, select the modified firmware and click on `Upgrade`
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 17]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+- After upgrade is done and you are able to reach the ONT via telnet, logon back and run these commands to change firmware version on both banks:
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:142]:
```suggestion
- Once the upgrade is done and you are able to reach the ONT via telnet, log back in and run these commands to change the firmware version on both banks:
```

File: _ont/ont-genexis-ft-g2110c.md [hunk 18]
```diff
@@ -65,7 +70,113 @@ This ONT supports dual boot.
...
+
+```sh
+# /etc/scripts/flash set OMCI_OLT_MODE 3
+OMCI_OLT_MODE=3
+# nv setenv sw_custom_version0=V6.0.10N14
+# nv setenv sw_custom_version1=V6.0.10N14
+```
+
+Reboot the ONT and check if version was correct applied with this command (take a look at `Version` field):
```
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:151]:
```suggestion
Reboot the ONT and check if the version was correctly applied with this command (take a look at the `Version` field):
```

File: _ont/ont-genexis-ft-g2110c.md (hunk 9 above)
Comment (nanomad) [inline _ont/ont-genexis-ft-g2110c.md:179]:
```suggestion
- Modified image (based on C-5.7.1-DEV3) has the Web UI enabled, but without an attached fiber the daemon will crash
```

**Review by nanomad — CHANGES_REQUESTED** [review]

**Review by nanomad — APPROVED** [review]

**Pull request merged**